llama.cpp

```diff
@@ -1575,13 +1502,10 @@ static bool llama_kv_cache_init(
     cache.cells.clear();
     cache.cells.resize(n_ctx);
 
-    cache.buf.resize(ggml_row_size(ktype, n_elements) + ggml_row_size(vtype, n_elements) + 2u*n_layer*ggml_tensor_overhead());
-    memset(cache.buf.data, 0, cache.buf.size);
```

Comment:
@ggerganov is this memset in the KV cache necessary? ggml-backend buffers don't have an equivalent of memset at the moment, do we need to add one? I tested removing the memset, and couldn't notice any obvious issues.

Reply:
I added this when we started padding the `kv_self.n`:

https://github.com/ggerganov/llama.cpp/blob/a7aee47b98e45539d491071b25778b833b77e387/llama.cpp#L6283-L6286

Although we are masking the padded values, my concern was that there could be NaNs if the KV cache is not initialized. We can remove the padding (it's optional), but I think in the long run it's better to have it as it will allow some optimizations in the attention.